src/Strings.h

```diff
@@ -472,14 +581,96 @@ template <
     typename UnaryPredicateT
 >
 void ParseNgramCharwbCopy(std::string const &input,
+                          std::vector<std::string> &intermediateValues,
                           UnaryPredicateT const &predicate,
                           size_t const ngramRangeMin,
                           size_t const ngramRangeMax,
                           std::function<void (IteratorT, IteratorT)> const &callback) {
 
     //copy string to trim
     std::string inputCopy(input);
-    ParseNgramCharwb(inputCopy, predicate, ngramRangeMin, ngramRangeMax, callback);
+    ParseNgramCharwb(inputCopy, intermediateValues, predicate, ngramRangeMin, ngramRangeMax, callback);
+}
+
+//to uniform so that all parse functions have same number of signature
```

Comment:
This doesn't feel right either. It isn't clear to me why these changes are necessary.